### ISSUE TITLE
Some RP-1 thoughts (mostly UI related)

### ISSUE THREAD

**Issue by Schnobs** [KSP-RO/RP-0#845]
Hello,

this is just a big dump of impressions.

Research. At any give time I have a lot of undone research in my KCT queue, which just isn't a suitable interface to handle it. Nevermind that you can rearrange things to be researched out of order, doing the second step before the first (which is a plain old bug), KCT merely provides you with a list of names devoid of any content or relationships. At the very least, it should be possible to rearrange the research queue from inside R&D so one can look at the tech tree at the same time. Some drag-and-drop when rearranging the list would also be helpful, as would be a second timer counting total time to unlock the second, third, _n_th entry in the list.

(actually, I have an inkling that the whole research paradigm with unlocking and researching individual nodes should be reconsidered, but don't have a proposal for a new system at hand)

Likewise, purchasing items is a major problem. You often cannot afford to just buy the whole contents of a tech node, and selective purchase isn't supported very well.  A whole new UI would be in order. Which should support a try-before-you-buy approach to rocket design, where you can assemble vessels from whatever parts, try them out in a sim, but ultimately only purchase what you actually want to use for reals.

As a corollary to the previous, I'd like to have better support for loading saved vessels. Real Fuels quietly demotes the engines to whatever models you have unlocked at the time you load the vessel (especially insidous when RCS thrusters get demoted to another propellant), proc avionics get reset in a similar manner. It's up to the player to remember what unlocks are needed to make the vessel work as originally designed.

Gameplay:

The fine-grained tech tree feels odd. Most games train you to expect every single unlock to be a meaningful breakthrough in it's own right, which just isn't the case in RP-0. One gets used to it, but at first there is a feeling of make-work. It's also strange to see a lot of time pass with very little progress. There's an impression that one *has* to do a lot of sounding rocket launches.

Tooling is a PITA. Contracts are all across the board with small and large payloads and altitudes, and every tiny variation in your rocket needs to be tooled or you'll face a preposterous increase in construction time. How tubes of varying length can be such an obstacle is beyond me. "We can make pieces of 30, 40 and 60 cm. Also 2.5m, 3m, and 3.2m. But a 50cm piece will require a whole new lathe, or you wait six weeks until we whittle it down by hand..."

Some kind of dV chart giving ballpark estimates of what you need to get to altitude X would come in handy.

I eventually made an oversized "Mother Of All Sounding Rockets" kind of thing which was supposed to provide for any contract the game would throw at me. Nevermind if that is intended player behaviour, it didn't even work because contracts kept raising the bar. Most annoying.

It might be nice to have an alternate entry point at about 1960.

There's some definite funneling towards doing an american space program. Getting an R-7-sized launchpad first isn't out of the question, but goes against the incentives.

I'm having trouble accounting for future part purchases. I tried, but quickly resorted to just keeping some cash in the bank and making things up as I go. Is that how it's supposed to be?

Tooling II: tried to be smart and design the first version of a rocket with a under-utilized bigger tank. You know, so it would still work with the next-generation engine of increased thrust and burn time. Only, what will be the burn time? That's a TF property and the RF mouseover won't tell me.

--
That's it for now. More might follow.

**Comment by ppboyle** [comment]
Keep in mind that RP-0/RP-1 is erring on the side of simulationism rather than smooth gameplay. Not to say that both can't/shouldn't be improved or that they're mutually exclusive, but it explains why some decisions are made.  There are a lot of other mods/mods suites for KSP that are more gamey.

For your last bit, though, mechjeb inside your VAB should have told you the burn time of your underfueled tank?  And middle clicking on an engine should give you the TF properties, I believe.

**Comment by Schnobs** [comment]
Yeah, more. Just did my first manned mission and wonder why I have to care about crew training, or crew resigning. That's middle managment and HR stuff. (also, yet another clumsy interface).

**Comment by Schnobs** [comment]
TL;DR: for the big picture, KCT needs to be thrown out and replaced with something more sensible, possibly something that's part of RP-0 proper. Don't know who could make that happen or when, so let's ignore this for now. I'll make a separate issue about training, tho.
